org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -235,6 +235,72 @@ public static boolean isOnSourcePath(IPath sourcePath, IJavaProject project) thr
 		return false;
 	}
 
+	public static boolean setInvisibleProjectOutputPath(IJavaProject javaProject, String outputPath, IProgressMonitor monitor) throws CoreException {
+		IProject project = javaProject.getProject();
+		if (isVisibleProject(project)) {
+			return false;
+		}
+
+		if (outputPath == null) {
+			outputPath = "";
+		} else {
+			outputPath = outputPath.trim();
+		}
+		
+		if (new org.eclipse.core.runtime.Path(outputPath).isAbsolute()) {
+			throw new CoreException(new Status(IStatus.ERROR, IConstants.PLUGIN_ID, "The output path must be a relative path to the workspace."));
+		}
+
+		if (StringUtils.isEmpty(outputPath)) {
+			// blank means using default output path
+			javaProject.setOutputLocation(project.getFolder("bin").getFullPath(), monitor);
+			return true;
+		}
+
+		outputPath = WORKSPACE_LINK + IPath.SEPARATOR + outputPath;
+		IPath outputFullPath = project.getFolder(outputPath).getFullPath();
+		IClasspathEntry[] existingEntries = javaProject.getRawClasspath();
+		List<IClasspathEntry> newEntries = new ArrayList<>();
+		for (IClasspathEntry entry : existingEntries) {
+			if (entry.getEntryKind() != IClasspathEntry.CPE_SOURCE) {
+				newEntries.add(entry);
+			} else {
+				IPath entryPath = entry.getPath();
+				if (entryPath.equals(outputFullPath)) {
+					throw new CoreException(new Status(IStatus.ERROR, IConstants.PLUGIN_ID,
+							"The output path cannot equal to source folder path, please provide a new path instead."));
```

Comment:
The output path cannot be equal to the source folder path, please provide a new path.